selene-lib/src/rules/limit_function_complexity.rs

```diff
@@ -0,0 +1,214 @@
+use super::*;
+use crate::ast_util::range;
+use std::convert::Infallible;
+
+
+use full_moon::{
+    ast::{self, Ast},
+    // node::Node,
+    visitors::Visitor,
+};
+
+use serde::Deserialize;
+
+#[derive(Clone, Copy, Deserialize)]
+pub struct LimitFunctionComplexityConfig {
+    maximum_complexity: u16,
+}
+
+impl Default for LimitFunctionComplexityConfig {
+    fn default() -> Self {
+        Self {
+            maximum_complexity: 100.to_owned(),
+        }
+    }
+}
+
+
+#[derive(Default)]
+pub struct LimitFunctionComplexityLint {
+    config: LimitFunctionComplexityConfig
+}
+
+impl Rule for LimitFunctionComplexityLint {
+    type Config = LimitFunctionComplexityConfig;
+    type Error = Infallible;
+
+    const SEVERITY: Severity = Severity::Warning;

```

Comment:
Don't do it yet but before I merge I'm going to replace this with:

```suggestion
    const SEVERITY: Severity = Severity::Allow;

```

...and make sure that works. Waiting on this so I can run selenology over everything first.